### DISCUSSION TITLE
Fake notification by Github Team

### DISCUSSION THREAD

**Discussion by grapts** [community/community#30251] [Other Feature Feedback, Questions, & Ideas]
Hello Everyone,

I received a notification from the GitHub Team  (enclosed) claiming that the token for my project would expire in 6 days. However, I was browsing through the GitHub documentation, and it seems the token expires only after 1 year. Does anyone know who sends such creepy fake notifications? 

Thanks!

Fake notification sent by Github Team:

----- Forwarded message from GitHub <noreply@github.com> -----
   Date: Sat, 20 Aug 2022 21:23:43 +0000 (UTC)
   From: GitHub <noreply@github.com>
Subject: [GitHub] Your personal access token is about to expire
     

Hi @grapts,

We noticed your personal access token "IBM Quantum Lab" with repo scope will expire in 6 days.

If this token is still needed, visit https://github.com/settings/tokens/908304509/regenerate to generate an equivalent.

If you run into problems, please contact support by visiting https://github.com/contact?tags=dotcom-accounts

Thanks,
The GitHub Team

----------------------------------------------------------------------------------
Further details on token expiration protocol can be found  in this link:

https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/token-expiration-and-revocation#token-revoked-by-the-user

**Comment by airtower-luna** [comment]
> However, I was browsing through the GitHub documentation, and it seems the token expires only after 1 year.

That's not correct as a general statement, you can choose the validity period when you create a token, and the default is *a lot shorter*. I just checked, and saw 30 days. You can check when your tokens actually expire at "Settings > Developer settings > Personal access tokens".

If you don't have such a token that's indeed suspicious, and you should report it to [GitHub Support](https://support.github.com/request).

And it's always smart to not log in from links in emails.

**Reply by grapts** [reply]
Here is the link to the GiHub Document stating "Token expired due to lack of use: 
GitHub will automatically revoke an OAuth token or personal access token when the token hasn't been used in one year."  

https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/token-expiration-and-revocation#token-revoked-by-the-user

Thank you for the information, it was very helpful. I definitely did not log in using any link in an email!

**Reply by airtower-luna** [reply]
Yes, that's if you create a token that's valid for a longer time ("doesn't expire" is possible, too) and then don't use it for a year. Naturally I don't know with which settings you created yours. :slightly_smiling_face: 

When I've created short-lived tokens I've always gotten that kind of notice about a week before they expired. Which is a bit weird if you intentionally create a token that's valid for 7 days, and then immediately get the notice. Yes, I know I set the token I just created to expire in a week. :sweat_smile: Not sure if they've changed that since.

> I definitely did not log in using any link in an email!

That's good. I could've probably phrased that better, it was just meant as a general warning, not implying you did. :slightly_smiling_face:

**Comment by DenverCoder1** [comment]
The documentation link you sent is for a completely different type of token. It says that *Oauth tokens* for unused apps will expire after a year. This means access to apps you've logged in to with your GitHub account.

The email message is telling you about the expiration of a repo scope *personal access token* which you must have created for your project. As airtower-luna said, the expiration of these is chosen by the user on creation of the token. The default is 30 day expiration, but you can choose for it to never expire as an option when creating the token.

You can visit the link in the email to regenerate the PAT (personal access token) to use for a longer period of time. The link is legitimate and you can navigate to it through your GitHub account settings.

You can see all your tokens and their expirations at <https://github.com/settings/tokens/>.

**Reply by grapts** [reply]
@DenverCoder1 - The documentation I sent is the only information I found online about tokens, if this information is not accurate it should be corrected asap. I know "Airluna", they have no clue about what they are talking about, therfore my response. Hopefully they will refrain from initiating uninvited chats in the future.

**Reply by DenverCoder1** [reply]
Here is some information in the documentation about Personal Access Tokens which is the kind your token is:
https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/creating-a-personal-access-token

And also the link you sent, but a few paragraphs higher on the page:
https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/token-expiration-and-revocation#token-revoked-after-reaching-its-expiration-date

> When you create a personal access token, we recommend that you set an expiration for your token. Upon reaching your token's expiration date, the token is automatically revoked

Both the links above and the paragraph you sent are correct but are talking about different methods of authentication.

I have seen airtower-luna around these forums elsewhere and they seem to know what they're talking about for the most part. There just seemed to be some confusion about the link referring to oauth authentication. Possibly you know someone else who goes by the same name? Either way I don't think your last reply to them was necessary.

**Comment by rathodtejal2792** [comment]
Can Anyone help , how to receive notification one day before on mail , if fine grained token is expire for repository ?
